contracts/scripts/testing.ts

```diff
@@ -0,0 +1,217 @@
+  const assertionConfirmedFilter = rollupContract.filters.AssertionConfirmed();
+
+  return {
+    sequencerContract,
+    rollupContract,
+    appendTxFilter,
+    assertionCreatedFilter,
+    assertionConfirmedFilter,
+  };
+}
+
+// Send a tx
+async function sendTx(sequencerSigner: any, toAddress: any, value: number) {
+  const nonce = await l2Provider.getTransactionCount(sequencerSigner.address);
+
+  const txData = {
+    to: toAddress,
+    value: value,
+    nonce: nonce,
+  };
+
+  const txResponse = await sequencerSigner.sendTransaction(txData);
+  await txResponse.wait();
+
+  const txReceipt = await l2Provider.getTransactionReceipt(txResponse.hash);
+  assert(txReceipt, "No tx on L2 blockchain");
+
+  return txResponse;
+}
+
+// Check logs
+async function checkLogs(name: string, contract: any, filter: any) {
+  const logs = await contract.queryFilter(filter);
+  assert(logs.length > 0, `No matching logs found for ${name}`);
+}
+
+// test Tx
+async function testTx(
+  sequencerSigner: any,
+  validatorSigner: any,
+  sequencerContract: any,
+  rollupContract: any,
+  appendTxFilter: any,
+  assertionCreatedFilter: any,
+  assertionConfirmedFilter: any,
+  toAddress: any,
+  value: any
+) {
+  const txResponse = await sendTx(sequencerSigner, toAddress, value);
+
+  await delay(60);
+  await checkLogs("appendTxFilter", sequencerContract, appendTxFilter);
+  await checkLogs(
+    "assertionCreatedFilter",
+    rollupContract,
+    assertionCreatedFilter
+  );
+  await checkLogs(
+    "assertionConfirmedFilter",
+    rollupContract,
+    assertionConfirmedFilter
+  );
+
+  return txResponse;
+}
+
+// New Test tx flow
+async function testTxs(toAddress: string, value: BigNumber) {
+  const { sequencerSigner, validatorSigner } = await setupSigners(
+    sequencerPrivateKeyPath,
+    validatorPrivateKeyPath
+  );
+
+  let data: string;
+  try {
+    data = fs.readFileSync(CONFIGURE_SYSTEM_PATH, "utf8");
+  } catch (err) {
+    console.error("Failed to read file:", err);
+    return;
+  }
+
+  const sequencerContractAddress: string = getSequencerAddress(data);
+  const rollupContractAddress: string = getRollupAddress(data);
+
+  console.log(`SEQUENCER_ADDR: ${sequencerContractAddress || "not found"}`);
+  console.log(`ROLLUP_ADDR: ${rollupContractAddress || "not found"}`);
+
+  const {
+    sequencerContract,
+    rollupContract,
+    appendTxFilter,
+    assertionCreatedFilter,
+    assertionConfirmedFilter,
+  } = initializeContracts(
+    sequencerContractAddress,
+    sequencerInboxJson.abi,
+    rollupContractAddress,
+    rollupJson.abi
+  );
+
+  for (let i = 0; i < 1; i++) {
+    const res = await testTx(
+      sequencerSigner,
+      validatorSigner,
+      sequencerContract,
+      rollupContract,
+      appendTxFilter,
+      assertionCreatedFilter,
+      assertionConfirmedFilter,
+      toAddress,
+      value
+    );
+  }
+}
+
+// Send multiple Txs
+async function sendMultipleTxs() {
```

Comment:
what is the difference between `testTxs` and `sendMultipleTxs`? why do you need both?

Reply:
Updated it to sendTxs for sending transaction